In [ ]:
#Ordenar los números de menor a mayor
x_i = [0.8797, 0.3884, 0.6289, 0.8750, 0.5999, 0.8589, 0.9996, 0.2415, 0.3808, 0.9606, 0.9848, 0.3469, 0.7977, 0.5844, 0.8147, 0.6431, 0.7387, 0.5613, 0.0318,
         0.7401, 0.4557, 0.1592, 0.8536, 0.8846, 0.3410, 0.1492, 0.8681, 0.5291, 0.3188, 0.5992, 0.9170, 0.2204, 0.5991, 0.5461, 0.5739, 0.3254, 0.0856, 0.2258,
         0.4603, 0.5027, 0.8376, 0.6235, 0.3681, 0.2088, 0.1525, 0.2006, 0.4720, 0.4272, 0.6360, 0.0954]

x_i.sort()

Calcular

$$D^+=\max_{1<i<n} \Bigg|\frac{i}{n}-x_i\Bigg|$$

In [ ]:
n = len(x_i)
D_mas = []
for i in range(n):
  D_i = abs(i/50-x_i[i])
  D_mas.append(D_i)

max(D_mas)

0.09879999999999997

Calcular

$$D^-=\max_{1<i<n} \Bigg|x_i-\frac{i-1}{n}\Bigg|$$

In [ ]:
D_min = []
for i in range(n):
  D_e = abs(x_i[i] - (i-1)/50)
  D_min.append(D_e)

max(D_min)

0.11879999999999996

Calcular

$$\max(D^+,D^-)$$

In [ ]:
max(max(D_mas), max(D_min))

0.11879999999999996

Tenemos que $d_{\alpha,n}$=0.1246.

Dado que $0.1188< 0.1246$, aceptamos la hipotesis.